mikeio1d/result_reader_writer/result_reader.py

```diff
@@ -177,6 +184,41 @@ def _add_node(self, node_id):
     def _add_catchment(self, catchment_id):
         self.data_subfilter.Catchments.Add(catchment_id)
 
+    def _setup_filter_time(self):
+        if self._time is None:
+            return
+        
+        period = self._make_period_from_time(self._time)
+        self.data_filter.Periods.Add(period)
+
+    def _make_period_from_time(self, time):
```

Comment:
I suggest to introduce a wrapper class, something like TimeFilter. Then there we could also handle indexed filtering etc.